libethereum/DatabasePaths.h

```diff
@@ -0,0 +1,27 @@
+// Aleth: Ethereum C++ client, tools and libraries.
+// Copyright 2014-2019 Aleth Authors.
+// Licensed under the GNU General Public License, Version 3.
+
+#pragma once
+
+#include "libdevcore/FixedHash.h"
+#include <boost/filesystem/path.hpp>
+
+namespace dev
+{
+namespace eth
+{
+namespace database_paths
+{
+bool databasePathsSet();
+void setDatabasePaths(boost::filesystem::path const& _rootPath, h256 const& _genesisHash);
+boost::filesystem::path rootPath();
```

Comment:
I think I would return const references from these, to avoid copying

Reply:
Somewhat confusing that we have `db::databasePath()` and now these. Should all these maybe better put into `DBFactory.h`?

Reply:
@gumb0 I considered that but DBFactory strikes me as more general purpose..it doesn’t include anything related to chains and such. Should I maybe rename the new namespace and files to something like BlockchainDatabasePaths to make the differences more clear?

Reply:
The namespace and files are fine, but then maybe it makes sense to move `db::setDatabasePath` to this file and/or merge with `setDatabasePaths`?

Isn't `rootPath` here basically the same as `db::databasePath` ?

Reply:
@gumb0 Yes, rootPath is the same as db::databasePath. After looking at this a little more I like your original suggestion of merging these together...I'll move the DatabasePaths functions out of that namespace and into the dev::db namespace.